django/test/runner.py

```diff
@@ -800,19 +800,13 @@ def partition_suite_by_type(suite, classes, bins, reverse=False):
     Tests of type classes[i] are added to bins[i],
     tests with no match found in classes are place in bins[-1]
     """
-    suite_class = type(suite)
-    if reverse:
-        suite = reversed(tuple(suite))
-    for test in suite:
-        if isinstance(test, suite_class):
-            partition_suite_by_type(test, classes, bins, reverse=reverse)
+    for test in iter_test_cases(suite, reverse=reverse):
+        for i in range(len(classes)):
+            if isinstance(test, classes[i]):
```

Comment:
Fan of `enumerate`?
```suggestion
        for i, klass in enumerate(classes):
            if isinstance(test, klass):
```

Reply:
I was avoiding doing any refactoring of `partition_suite_by_type()` to keep this PR focused just on `iter_test_cases()`. (I was actually planning to file a ticket on simplifying `partition_suite_by_type()` in a bunch of ways after this is merged. To give one example, you can actually dispense with the indices entirely and just use `zip()`.)